core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,22 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        if (analyzerSettings.getAsMap().containsKey("position_increment_gap")
+                && analyzerSettings.getAsMap().containsKey("position_offset_gap")){
+            throw new IllegalArgumentException("Custom Analyzer [" + name() + "] has both [position_offset_gap] and [position_increment_gap]");
+        }
+
+        int positionIncrementGap = 0;
+        if (Version.indexCreated(indexSettings).before(Version.V_2_0_0) ){
```

Comment:
Do we really need to ignore the setting in post 2.0 indexes? Why not just support both for a while? You already check above that both aren't specified.